src/tests/system/tests/test_gdm_passkey.py

```diff
@@ -0,0 +1,413 @@
+def test_gdm__passkey_login_remove_passkey_mapping(client: Client, ipa: IPA):
+    """
+    :title: Login via GDM fails when passkey mapping removed from user
+    :setup:
+        1. Configure SSSD for gdm-switchable-auth and pam_cert_auth
+        2. Start SSSD
+        3. Start virtual passkey service
+        4. Add user to IPA and set auth_type to passkey
+        5. Register passkey with IPA user
+        6. Remove user passkey mapping from IPA
+    :steps:
+        1. Login through GDM using Passkey with PIN
+    :expectedresults:
+        1. Login unsuccessful and user prompted for password
+    :customerscenario: False
+    """
+    testuser = "pkgdmuser1"
+    pin = "123456"
+
+    # Configure SSSD and vfido
+    client_setup_for_passkey(client, ipa, pin=pin)
+
+    # Add IPA User
+    ipa.user(testuser).add(user_auth_type="passkey")
+
+    # Register passkey with IPA User
+    ipa.user(testuser).passkey_add_register(client=client, pin=pin, virt_type="vfido")
+
+    result = ipa.user(testuser).get(["ipapasskey"])
+    if result is not None:
+        ipa.user(testuser).passkey_remove(result["ipapasskey"][0])
+    else:
+        raise ValueError(f"ipa.user({testuser}) passkey mapping not found")
+
+    # Login through GDM with PIN
+    client.gdm.click_on("listed?")
+    client.gdm.kb_write(testuser)
+    client.gdm.kb_send("tab")
+    assert client.gdm.assert_text("Password"), "User was not prompted for Password as expected!"
+
+
+@pytest.mark.builtwith(client=["gdm", "passkey", "vfido"])
+@pytest.mark.topology(KnownTopology.BareIPA)
+def test_gdm__passkey_login_with_unregistered_mapping(client: Client, ipa: IPA):
+    """
+    :title: Login via GDM fails with unregistered passkey mapping
+    :setup:
+        1. Configure SSSD for gdm-switchable-auth and pam_cert_auth
+        2. Start SSSD
+        3. Start virtual passkey service
+        4. Add user to IPA and set auth_type to passkey
+        5. Register passkey with IPA user
+        6. Remove user passkey mapping from IPA
+        7. Add bad passkey mapping to user in IPA
+    :steps:
+        1. Login through GDM using Passkey with PIN
+    :expectedresults:
+        1. Login unsuccessful and user is prompted again for PIN
+    :customerscenario: False
+    """
+    testuser = "pkgdmuser1"
+    pin = "123456"
+    bad_mapping = (
+        "passkey:oMpNJz5y4YLCB48dJGmAJQ==,"
+        "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEEd6cT8vlg7JHt6VJ"
+        "/NfE7TW+T/NrbNLZJdK2qewFUMicptYaD8FWDWYboeKPt17ukFRnDZj2VrDx70UYMMsXFA=="
+    )
+
+    # Configure SSSD and vfido
+    client_setup_for_passkey(client, ipa, pin=pin)
+
+    # Add IPA User
+    ipa.user(testuser).add(user_auth_type="passkey")
+
+    # Register passkey with IPA User
+    ipa.user(testuser).passkey_add_register(client=client, pin=pin, virt_type="vfido")
+
+    # Remove passkey mapping
+    result = ipa.host.conn.run("ipa user-show pkgdmuser1 --raw |grep ipapasskey:|awk '{print $2}'")
```

Comment:
IIRC I already opened a comment about this one and you closed it. I think you lost your progress there because I can still see `ipa user-show` instead of the test API